UISAUTCOMP-95 Use first page Browse query for facet requests.

BogdanDenis opened the pull request:
## Description
Fix Facet counts changing when going Next or Prev page in Browse
Use first page query to fetch Browse facets

## Approach
`useAuthoritiesBrowse` returns two query strings: `query` and `firstPageQuery`.
`query` describes search parameters that are used for browsing results for current page.
`firstPageQuery` describes search parameters that were used for browsing results for the initial page (page with anchor).
For facet requests we need to use `firstPageQuery` because if we navigate to next or previous page - the search parameters will change and we will get different facet counts.

For example, user browses `Twain, Mark`. The first page will contain the anchor and some other similar results. If the user then goes to see the previous page - Browse request will contain a parameter `headingRef < "TW, obscure Englishman"`. By using this parameter in facets we will lose the context of the initial browse search and so the facet counts will be different

![image](https://github.com/folio-org/stripes-authority-components/assets/19309423/466cfe95-aa39-43e4-bdcf-9286ce61d731)


## Screenshots

https://github.com/folio-org/stripes-authority-components/assets/19309423/8d802b7e-b8af-4818-b2b5-5466ee24bb40


## Issues
[UISAUTCOMP-95](https://issues.folio.org/browse/UISAUTCOMP-95)

## Related PRs
https://github.com/folio-org/ui-marc-authorities/pull/333
https://github.com/folio-org/ui-plugin-find-authority/pull/88

zburke reviewed (commented):
I'm glad `firstPageQuery` works, but, sorry, I have no idea how it works and there is no documentation in code or in a README describing this prop. How/why is `firstPageQuery` different than `query`? Given `useAuthoritiesBrowse` defines both, why isn't `query` necessary in `AuthoritiesSearchPane` when you're navigating past the first page? 

Is this component exported at the package-level? Adding a new required prop is a breaking change if so.

BogdanDenis commented:
Hey @zburke, thanks for review. I've changed the implementation slightly to not need a new prop. Also added some documentation on the hook and tried to explain how this fix works (hopefully well enough because Browsing can be tough to wrap your head around IMO)
